### DISCUSSION TITLE
Create a library for VTK mesh export

### DISCUSSION THREAD

**Discussion by germa89** [ansys/pydpf-core#421] [Polls]
Hi all,

It seems that capability of convert ANSYS mesh to VTK is important in DPF and PyMAPDL libraries.

These two libraries seem to use similar code:
- PyMAPDL (using legacy reader): https://github.com/pyansys/pymapdl-reader/blob/7341032069a0ddd9f4e21e5d22d888920b90e9bd/ansys/mapdl/reader/mesh.py#L146
- DPF: https://github.com/pyansys/pydpf-core/blob/master/ansys/dpf/core/vtk_helper.py

Surely, the code is not entirely the same, but I think we could reuse most of them in both libraries. In fact, some future libraries might reuse it too.

Therefore, I would propose to externalize this capability to a different library (something like ``ansys.helpers.vtk_export`` ?).

This will help inhttps://github.com/pyansys/pymapdl/pull/1299

**Notes**
I'm keeping the question and answers simple, but I'm open to heard other options/alternatives.

**Comment by germa89** [comment]
Pinging @PProfizi @akaszynski @rlagha @koubaa @cbellot000 for visibility.

**Reply by akaszynski** [reply]
PRIME needs this as well. Let's bring this up at the next weekly. @germa89, please pencil this in.

**Reply by germa89** [reply]
Roger!

**Comment by koubaa** [comment]
DPF has a concept of an abstract mesh, which can handle different mesh providers.  abstract mesh -> vtk would be generally useful, perhaps as a dpf operator

**Reply by germa89** [reply]
The feature  (this part https://github.com/pyansys/pydpf-core/blob/master/ansys/dpf/core/vtk_helper.py) could be wrapped in an operator if needed. I guess that should be fine then?

**Reply by cbellot000** [reply]
@koubaa that's right, but we already have an Operator name "vtk_export" which takes a IMeshedRegion (equivalent to abstract mesh) and write it into a vtk file but maybe what you need is not a file right?

**Reply by koubaa** [reply]
@cbellot000 We need the raw vtk polydata, edges, etc as arrays - not a file. If its possible to create an IMeshedRegion from arrays of nodes and element connectivity (I didn't find an example) then we can create an IMeshedRegion on the fly from the mapdl model without any result file.

**Reply by cbellot000** [reply]
@koubaa that definitely possible, @PProfizi I think you worked on example, could you please give us the link?

**Reply by PProfizi** [reply]
@cbellot000 Are you referring to creating a MeshedRegion from scratch using setters?
I do have a PR started to make an example for this, yet it is not finished. See [here](https://github.com/pyansys/pydpf-core/pull/366).
